src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -76,11 +92,45 @@ private void setSubmissionData(FilingApi filing, OverseasEntitySubmissionDto sub
         data.put(MANAGING_OFFICERS_INDIVIDUAL_FIELD, submissionDto.getManagingOfficersIndividual());
         data.put(MANAGING_OFFICERS_CORPORATE_FIELD, submissionDto.getManagingOfficersCorporate());
         data.put(BENEFICIAL_OWNERS_STATEMENT, submissionDto.getBeneficialOwnersStatement());
-        filing.setData(data);
 
-        setDescriptionFields(filing);
+        var logMap = new HashMap<String, Object>();
```

Comment:
How about creating this 'logMap' just once, in the `setFilingApiData()` method and passing it into this method and `setPaymentData()` below?

Reply:
will do

Reply:
updated